tests/baselines/reference/privateNameInTypeQueryImport.errors.txt

```diff
@@ -0,0 +1,14 @@
+tests/cases/conformance/classes/members/privateNames/main.ts(3,43): error TS2694: Namespace '"tests/cases/conformance/classes/members/privateNames/lib".Bar' has no exported member '#m'.
+
+
+==== tests/cases/conformance/classes/members/privateNames/main.ts (1 errors) ====
+    export class Foo {
+      static #m = 1;
+      static #f(x: typeof import("./lib").Bar.#m) {}
+                                              ~~
+!!! error TS2694: Namespace '"tests/cases/conformance/classes/members/privateNames/lib".Bar' has no exported member '#m'.

```

Comment:
This error seems to be wrong, since `Bar` is `Foo` and thus it has the `#m` property.

Reply:
I can't tell if it's completely correct or not.
In this PR, I convert SyntaxError into TypeError.
Maybe it should be counted as "external" access.

>
> According to the latest nightly for TS, `import(..).X.#priv` is technically syntactically valid, but it will never be semantically valid because it counts as "external" access to the private member, which is disallowed.
...

_Originally posted by @bradzacher in https://github.com/babel/babel/issues/14454#issuecomment-1100788711_

Reply:
I will try to add more test cases to determine which case is more reasonable and self-consistent.

Reply:
I think a comparable way to test this would be to stick `private` onto the non-`#` one and see what happens.

Reply:
> I think a comparable way to test this would be to stick `private` onto the non-`#` one and see what happens.

I didn't manage to do it.
At first I tried to alias `type Bar = typeof import("./lib").Bar`, then I found I cannot use `Bar.#m`.
` typeof import("./lib").Bar.#m` does not imply `(typeof import("./lib").Bar).#m`.
it means `typeof (import("./lib").Bar.#m)`.

What do you think about it?

Reply:
For `private`, I just meant to write something like:

```ts
// @strict: true
// @target: esnext

// @filename: main.ts
export class Foo {
  private static m = 1;
  static #f(x: typeof import("./lib").Bar.m) {}
}

// @filename: lib.ts
export { Foo as Bar } from "./main";
```

Where the static thing is "private" via TS, not via ECMAScript's private identifiers. I was replying on my phone so couldn't write it myself, sorry about that.

Reply:
Regarding the precedence, that's #19707 (mentioned in #48640). All of this is getting really hairy and inconsistent, though.

Reply:
The AST should be changed. This PR is blocked by #48640.